### PR TITLE
Create install class and unit tests

### PR DESCRIPTION
**Pull request by dhollinger**

### CODE REVIEW

File: .fixtures.yml [hunk 1]
```diff
@@ -2,5 +2,7 @@
 # See https://github.com/puppetlabs/puppetlabs_spec_helper#using-fixtures for details
 ---
 fixtures:
-  forge_modules:
-#     stdlib: "puppetlabs/stdlib"
+  repositories:
+    stdlib: "https://github.com/puppetlabs/puppetlabs-stdlib.git"
+  symlinks:
+    puppet_webhook: "#{source_dir}"
```
Comment (bastelfreak) [inline .fixtures.yml]: please add the missing newline.

File: spec/spec_helper_acceptance.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+RSpec.configure do |c|
+  # Project root
+  module_root = File.expand_path(File.join(File.dirname(__FILE__), '..'))
+
+  # Configure all nodes in nodeset
+  c.before :suite do
+    # Install module and deps
+    hosts.each do |host|
+      install_dev_puppet_module_on(host, source: module_root, module_name: 'autofs',
```
Comment (bastelfreak) [inline spec/spec_helper_acceptance.rb]: I try to always use:
install_module_on(hosts)
install_module_dependencies_on(hosts)
which installs the modules + all dependencies from metadata.json

File: .fixtures.yml (hunk 1 above)
Comment (alexjfisher) [inline .fixtures.yml]: I think the symlink is created by default these days.

Comment (alexjfisher) [reply .fixtures.yml]: https://github.com/puppetlabs/puppetlabs_spec_helper/blob/master/CHANGELOG.md#120---2016-08-23
```
Automatically symlink the module directory, if no symlink fixtures are specified.
```

File: spec/classes/puppet_webhook_spec.rb [hunk 3]
```diff
@@ -6,6 +6,8 @@
       let(:facts) { os_facts }
 
       it { is_expected.to compile }
```
Comment (ekohl) [inline spec/classes/puppet_webhook_spec.rb]: I like to use `compile.with_all_deps`

File: manifests/init.pp [hunk 4]
```diff
@@ -3,10 +3,16 @@
...
 # @example
 #   include puppet_webhook
 #
-class puppet_webhook {
+# @param pkg_provider Which provider to install puppet_webhook into
+#
+class puppet_webhook(
+  Enum['gem', 'puppet_gem'] $pkg_provider,
+) {
   contain 'puppet_webhook::install'
```
Comment (ekohl) [inline manifests/init.pp]: Should there be chaining from install -> config?

File: .sync.yml [hunk 5]
```diff
@@ -1,4 +1,25 @@
...
+        platforms: ruby
+      - gem: 'puppet-module-win-system-r#{minor_version}'
+        platforms:
+          - mswin
+          - mingw
+          - x64_mingw
+      - gem: beaker
+        version: '~> 3.13'
+        from_env: BEAKER_VERSION
+      - gem: beaker-abs
```
Comment (ekohl) [inline .sync.yml]: Do we really need this gem? I don't really see us using this.

File: spec/classes/puppet_webhook_spec.rb (hunk 3 above)
Comment (dhollinger) [reply spec/classes/puppet_webhook_spec.rb]: This is generated by PDK. I haven't changed this yet as I don't want failing tests on code that's not done yet (i.e. `puppet_webhook::config`. There shouldn't be any failures, but rather be safe than sorry.

That said, I fully intend to add that, but it'll be closer to initial release than as part of the install class PR.

File: manifests/init.pp (hunk 4 above)
Comment (dhollinger) [reply manifests/init.pp]: You don't need direct chaining anymore, there will be resource relationships setup where the files in `config` will require `Package['puppet_webhook']`. The contain will ensure that both classes are contained within the init class and aware of each other.

I should note that the config class has no code in it yet, the resource relationships will be setup when I start working on those

File: .sync.yml (hunk 5 above)
Comment (dhollinger) [reply .sync.yml]: Just pulled in from PDK templates. I'll fine-tune that later

File: .fixtures.yml (hunk 1 above)
Comment (dhollinger) [reply .fixtures.yml]: Did not know that, I'll remove it then

File: data/common.yaml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+---
+puppet_webhook::pkg_provider: 'puppet_gem'
```
Comment (bastelfreak) [inline data/common.yaml]: please add the missing newline

File: hiera.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+defaults:
+  data_hash: yaml_data
+  datadir: data
+hierarchy:
+  - name: "OS Family - Major Version"
+    path: "%{facts.os.family}-%{facts.os.release.major}.yaml"
+  - name: "OS Family"
+    path: "%{facts.os.family}.yaml"
+  - name: "Common values"
+    path: "common.yaml"
```
Comment (bastelfreak) [inline hiera.yaml]: please add the missing newline

File: spec/spec_helper_acceptance.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
+
+require 'beaker-rspec/spec_helper'
+require 'beaker-rspec/helpers/serverspec'
+require 'beaker/puppet_install_helper'
+require 'beaker/module_install_helper'
+
+run_puppet_install_helper unless ENV['BEAKER_provision'] == 'no'
```
Comment (bastelfreak) [inline spec/spec_helper_acceptance.rb]: We should probably add this to our modulesync_config rather sooner than later.

Comment (dhollinger) [reply spec/spec_helper_acceptance.rb]: For the other modules you mean? If this is going to be a PDK compat module, we can't use modulesync as well.

File: data/common.yaml (hunk 6 above)
Comment (ekohl) [reply data/common.yaml]: Nit: now there's an empty line

File: spec/spec_helper_acceptance.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+
```
Comment (ekohl) [inline spec/spec_helper_acceptance.rb]: Nit: can you remove this empty line?

Comment (dhollinger) [reply spec/spec_helper_acceptance.rb]: This should now be fixed.

File: data/common.yaml (hunk 6 above)
Comment (dhollinger) [reply data/common.yaml]: This should be fixed now

Comment (alexjfisher) [reply data/common.yaml]: Build is still failing. Did you push the fix?

File: spec/spec_helper_acceptance.rb (hunk 9 above)
Comment (alexjfisher) [reply spec/spec_helper_acceptance.rb]: This blank line is still failing the build I think.

Comment (dhollinger) [reply spec/spec_helper_acceptance.rb]: @alexjfisher it should be fixed now.... a month later

**Pull request merged**